src/d3/bar-chart.d3.ts

```diff
@@ -0,0 +1,138 @@
+import * as d3 from 'd3';
+import { XYChartD3 } from './xy-chart.d3';
+import { BarChartDatum } from '../components/bar-chart/bar-chart.component';
+import { CategoricalPoint } from '../datasets/metas/types';
+
+export interface LegendItemStyle {
+  color: string;
+  opacity: number;
+}
+
+export class BarChartD3 extends XYChartD3<CategoricalPoint, BarChartDatum> {
+  static defaultLegendItemStyle: LegendItemStyle = {
+    color: BarChartD3.primaryColor,
+    opacity: 1,
+  };
+
+  protected scaleX: d3.ScaleLinear<number, number>;
+  protected scaleY: d3.ScaleBand<string>;
+  protected xAxis: d3.Axis<number>;
+  protected yAxis: d3.Axis<string>;
+
+  protected activePointCircle: d3.Selection<SVGCircleElement, unknown, null, undefined>;
+
+  protected renderData() {
+    super.renderData();
+  }
+
+  x(value: number) {

```

Comment:
Change to get function? Here and below

Or are these getBarWidth and getBarHeight functions?

Reply:
Can get function take arguments?

`x` and `y` are more readable in the context of these d3 classes. Since we define members of class in `scaleX`, `xAxis`, `xAxisG` instead of `scaleWidth` or `scaleHorizontal`, calling them `x` and `y` here is more straightforward that these variables are related.